### ISSUE TITLE
upgrader not working with Flutter 2.0

### ISSUE THREAD

**Issue by NTMS2017** [larryaasen/upgrader#93]
After clean install flutter 2.0 the upgrade plugin is gives an error while running the app.
First I use upgrader: ^2.8.2 and got an error than I update the plugin to .0.0-nullsafety.2 and I got plugin installation error.

**upgrader: ^2.8.2 running app Error:**
```
Xcode's output:
↳
    ../flutter/.pub-cache/hosted/pub.dartlang.org/upgrader-2.8.2/lib/src/upgrade_messages.dart:76:48: Error: No named parameter with the name 'nullOk'.
          locale = Localizations.localeOf(context, nullOk: true);
                                                   ^^^^^^
    ../flutter/packages/flutter/lib/src/widgets/localizations.dart:413:17: Context: Found this candidate, but the arguments don't match.
      static Locale localeOf(BuildContext context) {
                    ^^^^^^^^
    ../flutter/.pub-cache/hosted/pub.dartlang.org/upgrader-2.8.2/lib/src/upgrader.dart:392:48: Error: No named parameter with the name 'nullOk'.
          locale = Localizations.localeOf(context, nullOk: true);
                                                   ^^^^^^
    ../flutter/packages/flutter/lib/src/widgets/localizations.dart:413:17: Context: Found this candidate, but the arguments don't match.
      static Locale localeOf(BuildContext context) {
                    ^^^^^^^^

    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

**upgrader: ^3.0.0-nullsafety.2 plugin install error:**
```
Because upgrader >=3.0.0-nullsafety.2 depends on package_info ^2.0.0 and vakiflarbank_cep depends on package_info ^0.4.3+4, upgrader >=3.0.0-nullsafety.2 is forbidden.
So, because vakiflarbank_cep depends on upgrader ^3.0.0-nullsafety.2, version solving failed.
pub get failed (1; So, because vakiflarbank_cep depends on upgrader ^3.0.0-nullsafety.2, version solving failed.)
```

**flutter doctor -v:**
```
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.0.1 at /Users/**********/flutter
    • Framework revision c5a4b4029c (2 days ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/**********/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/**********/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 54.0.3
    • Dart plugin version 203.7759

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.82

```

**Comment by larryaasen** [comment]
Since vakiflarbank_cep depends on package_info ^0.4.3+4 and upgrader >=3.0.0-nullsafety.2 depends on package_info ^2.0.0, you need to upgrade your version of package_info to the null safety version. Maybe vakiflarbank_cep has a null safety version now.

**Comment by larryaasen** [comment]
Closed due to lack of activity. You can open a new issue if you have more questions.
